src/Servers/Kestrel/Core/src/CertificateLoader.cs

```diff
@@ -36,13 +36,19 @@ public static X509Certificate2 LoadFromStoreCert(string subject, string storeNam
                 {
                     store.Open(OpenFlags.ReadOnly);
                     storeCertificates = store.Certificates;
-                    var foundCertificates = storeCertificates.Find(X509FindType.FindBySubjectName, subject, !allowInvalid);
-                    foundCertificate = foundCertificates
+                    foreach (var certificate in storeCertificates.Find(X509FindType.FindBySubjectName, subject, !allowInvalid)
                         .OfType<X509Certificate2>()
                         .Where(IsCertificateAllowedForServerAuth)
                         .Where(DoesCertificateHaveAnAccessiblePrivateKey)
-                        .OrderByDescending(certificate => certificate.NotAfter)
-                        .FirstOrDefault();
+                        .OrderByDescending(certificate => certificate.NotAfter))
+                    {
+                        foundCertificate = certificate;
+
+                        if (foundCertificate.GetNameInfo(X509NameType.SimpleName, true).Equals(subject, StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Why ignore case?

Reply:
Current implementation before this PR also case insensitive.

>          // Summary:
>         //     The findValue parameter for the System.Security.Cryptography.X509Certificates.X509Certificate2Collection.Find(System.Security.Cryptography.X509Certificates.X509FindType,System.Object,System.Boolean)
>         //     method must be a string representing the subject name of the certificate. This
>         //     is a less specific search than that provided by the System.Security.Cryptography.X509Certificates.X509FindType.FindBySubjectDistinguishedName
>         //     enumeration value. Using the System.Security.Cryptography.X509Certificates.X509FindType.FindBySubjectName
>         //     value, the System.Security.Cryptography.X509Certificates.X509Certificate2Collection.Find(System.Security.Cryptography.X509Certificates.X509FindType,System.Object,System.Boolean)
>         //     method performs a case-insensitive string comparison using the supplied value.
>         //     For example, if you pass "MyCert" to the System.Security.Cryptography.X509Certificates.X509Certificate2Collection.Find(System.Security.Cryptography.X509Certificates.X509FindType,System.Object,System.Boolean)
>         //     method, it will find all certificates with the subject name containing that string,
>         //     regardless of other subject values. Searching by distinguished name is a more
>         //     precise search.
>         FindBySubjectName,

Reply:
Yes, but the goal now is to do an exact match if possible - if there is a `mycert` and a `MyCert` present, and the user looks for `MyCert`, it should always pick the one that matches most closely. @BrennanConroy do you agree?

Reply:
I doubt there are a great chances for systems to have same certificate name in different casing, but as a valid use case could be having `mycert` certificate and user search for `MyCert`. Users may be surprised to not found one. Moreover this would confuse users even more when the substring search allow them to find certificates perfectly.